src/main/java/com/pusher/client/crypto/nacl/SecretBoxOpener.java

```diff
@@ -33,19 +36,15 @@ a copy of this software and associated documentation files (the "Software"),
     private byte[] key;
 
     public SecretBoxOpener(byte[] key) {
-        // TODO: add a a little preconditions lib/class for that
-        if (key == null) throw new IllegalArgumentException("null key passed");
-        if (key.length != 32) {
-            throw new IllegalArgumentException("key should be 32B, is: " + key.length + "B");
-        }
+        checkNotNull(key, "null key passed");
+        checkArgument(key.length == 32, "key length should be 32 bytes, but is " +
```

Comment:
nit: should -> must if we're failing people for it not being what we expect.

Reply:
I'm not sure if that doesn't sound too strong and if "should" is not more common for such messages (after all it also can be used to express obligation). @mdpye thoughts? Changed it for feedback: bb925cd